functions/supplier/zeromq-supplier/src/main/java/org/springframework/cloud/fn/supplier/zeromq/ZeroMqSupplierProperties.java

```diff
@@ -0,0 +1,153 @@
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import org.hibernate.validator.constraints.Range;
+import org.zeromq.SocketType;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.util.Assert;
+import org.springframework.validation.annotation.Validated;
+
+/**
+ *
+ * @author Daniel Frey
+ * @since 3.1.0
+ */
+@ConfigurationProperties("zeromq.supplier")
+@Validated
+public class ZeroMqSupplierProperties {
+
+	private static final SocketType DEFAULT_SOCKET_TYPE = SocketType.SUB;
+	private static final Duration DEFAULT_CONSUME_DELAY = Duration.ofSeconds(1);
+	private static final List<SocketType> VALID_SOCKET_TYPES =
+			Arrays.asList(SocketType.PAIR, SocketType.PULL, SocketType.SUB);
+
+	private final AtomicInteger bindPort = new AtomicInteger();
+
+	/**
+	 * The {@link SocketType} the connection should make.
+	 *
+	 * Allowable values:
+	 * - PAIR
+	 * - PULL
+	 * - SUB
+	 */
+	private SocketType socketType = DEFAULT_SOCKET_TYPE;
+
+	/**
+	 * Connection URL for {@link org.zeromq.ZMQ.Socket#connect(String)}.
+	 * Mutually exclusive with the {@link #bindPort}.
+	 */
+	private String connectUrl;
+
+	/**
+	 * A {@link Duration} to delay consumption when no data received.
+	 */
+	private Duration consumeDelay = DEFAULT_CONSUME_DELAY;
+
+	/**
+	 * Topics the {@link SocketType#SUB} socket is going to use for subscription.
+	 * It is ignored for all other {@link SocketType}s supported.
+	 */
+	private String[] topics = {""};
+
+	/**
+	 * @param socketType the {@link SocketType} to establish,
+	 *                      defaults to {@link #DEFAULT_SOCKET_TYPE} if empty
+	 */
+	@NotNull(message = "'socketType' is required")
+	public void setSocketType(SocketType socketType) {
+		Assert.state(VALID_SOCKET_TYPES.contains(socketType),
+				() -> "'socketType' can only be one of the: " + VALID_SOCKET_TYPES);
+		this.socketType = socketType;
+	}
+
+	public SocketType getSocketType() {
+		return socketType;
+	}
+
+	@NotEmpty(message = "connectUrl is required like tcp://server:port")
+	public String getConnectUrl() {
+		return connectUrl;
+	}
+
+	/**
+	 *
+	 * @param connectUrl The ZeroMQ TCP server connect url
+	 *
+	 * @see org.springframework.integration.zeromq.inbound.ZeroMqMessageProducer#setConnectUrl(String)
+	 */
+	@NotEmpty(message = "'connectUrl' must not be empty")
+	public void setConnectUrl(String connectUrl) {
+		this.connectUrl = connectUrl;
+	}
+
+	public Duration getConsumeDelay() {
+		return consumeDelay;
+	}
+
+	/**
+	 * @param bindPort The TCP Port to bind to on all interfaces
+	 *
+	 * @see org.springframework.integration.zeromq.inbound.ZeroMqMessageProducer#setBindPort(int)
+	 */
+	@Range(min = 0, message = "'bindPort' must not be zero or negative")
+	public void setBindPort(int bindPort) {
+		this.bindPort.set(bindPort);
```

Comment:
I don't think we need an `AtomicInteger` on the configuration properties level...
It is just an internal logic of the channel adapter.